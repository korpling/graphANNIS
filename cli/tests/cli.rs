use assert_cmd::prelude::*;
use insta::Settings;
use insta_cmd::assert_cmd_snapshot;
use serial_test::serial;
use std::process::Command;

fn standard_filter() -> Settings {
    let mut settings = insta::Settings::clone_current();
    // Remove any color ASCII codes
    settings.add_filter("\x1b", "");
    settings.add_filter("\\[[0-9]+m", "");

    // Filter out the time stamps
    settings.add_filter("[0-9]+:[0-9]+:[0-9]+ ", "12:00:00");
    // The loaded and also total available RAM size can vary
    settings.add_filter("[0-9.]+[MG]B / [0-9.]+[MG]B", "100MB / 300MB");
    // The loading and time can vary
    settings.add_filter("in [0-9]+ ms", "in 10 ms");
    settings
}

#[test]
#[serial]
fn show_corpus_info() -> Result<(), Box<dyn std::error::Error>> {
    let mut cmd = Command::cargo_bin("annis")?;

    cmd.arg("../graphannis/tests/data/")
        .arg("-c")
        .arg("corpus sample-disk-based-3.3")
        .arg("-c")
        .arg("preload")
        .arg("-c")
        .arg("info");

    let settings = standard_filter();
    settings.bind(|| assert_cmd_snapshot!(cmd));

    Ok(())
}

#[test]
#[serial]
fn list_corpora_not_loaded() -> Result<(), Box<dyn std::error::Error>> {
    let mut cmd = Command::cargo_bin("annis")?;

    cmd.arg("../graphannis/tests/data/").arg("-c").arg("list");

    let settings = standard_filter();
    settings.bind(|| assert_cmd_snapshot!(cmd));

    Ok(())
}

#[test]
#[serial]
fn list_corpora_fully_loaded() -> Result<(), Box<dyn std::error::Error>> {
    let mut cmd = Command::cargo_bin("annis")?;

    cmd.arg("../graphannis/tests/data/")
        .arg("-c")
        .arg("corpus sample-disk-based-3.3")
        .arg("-c")
        .arg("preload")
        .arg("-c")
        .arg("list");

    let settings = standard_filter();
    settings.bind(|| assert_cmd_snapshot!(cmd));

    Ok(())
}

#[test]
#[serial]
fn list_corpora_partially_loaded() -> Result<(), Box<dyn std::error::Error>> {
    let mut cmd = Command::cargo_bin("annis")?;

    cmd.arg("../graphannis/tests/data/")
        .arg("-c")
        .arg("corpus sample-disk-based-3.3")
        .arg("-c")
        .arg("count tok")
        .arg("-c")
        .arg("list");

    let settings = standard_filter();
    settings.bind(|| assert_cmd_snapshot!(cmd));

    Ok(())
}
