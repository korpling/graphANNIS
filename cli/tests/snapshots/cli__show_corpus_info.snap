---
source: cli/tests/cli.rs
info:
  program: annis
  args:
    - "../graphannis/tests/data/"
    - "-c"
    - corpus sample-disk-based-3.3
    - "-c"
    - preload
    - "-c"
    - info
---
success: true
exit_code: 0
----- stdout -----
12:00:00[INFO] Loaded corpus sample-disk-based-3.3
12:00:00[INFO] Corpus cache after preloading sample-disk-based-3.3: 100MB / 300MB - loaded corpora [sample-disk-based-3.3]
12:00:00[INFO] Preloaded corpus in 10 ms
Status: "fully loaded"
Token search shortcut possible: true
------------
Component Coverage//: 0 annnotations
Stats: nodes=92, avg_fan_out=2.17, max_fan_out=11, fan_out_99%=11, inv_fan_out_99%=9, max_depth=1
Implementation: DiskAdjacencyListV1
Status: "fully loaded"
------------
Component Coverage/annis/: 0 annnotations
Stats: nodes=0, avg_fan_out=0.00, max_fan_out=0, fan_out_99%=0, inv_fan_out_99%=0, max_depth=1, tree
Implementation: DiskAdjacencyListV1
Status: "fully loaded"
------------
Component Coverage/default_ns/: 0 annnotations
Stats: nodes=56, avg_fan_out=0.93, max_fan_out=10, fan_out_99%=10, inv_fan_out_99%=2, max_depth=1
Implementation: DiskAdjacencyListV1
Status: "fully loaded"
------------
Component Coverage/annis/inherited-coverage: 0 annnotations
Stats: nodes=0, avg_fan_out=0.00, max_fan_out=0, fan_out_99%=0, inv_fan_out_99%=0, max_depth=1, tree
Implementation: DiskAdjacencyListV1
Status: "fully loaded"
------------
Component Dominance/syntax/: 0 annnotations
Stats: nodes=92, avg_fan_out=0.96, max_fan_out=3, fan_out_99%=3, inv_fan_out_99%=1, max_depth=9, tree
Implementation: PrePostOrderO16L8V1
Status: "fully loaded"
------------
Component Pointing/default_ns/anaphoric: 0 annnotations
Stats: nodes=8, avg_fan_out=0.50, max_fan_out=1, fan_out_99%=1, inv_fan_out_99%=1, max_depth=1, tree
Implementation: DiskPathV1_D15
Status: "fully loaded"
------------
Component Ordering/annis/: 0 annnotations
Stats: nodes=44, avg_fan_out=0.91, max_fan_out=1, fan_out_99%=1, inv_fan_out_99%=1, max_depth=10, tree
Implementation: DiskPathV1_D15
Status: "fully loaded"
------------
Component LeftToken/annis/: 0 annnotations
Stats: nodes=92, avg_fan_out=0.65, max_fan_out=1, fan_out_99%=1, inv_fan_out_99%=3, max_depth=1
Implementation: DiskPathV1_D15
Status: "fully loaded"
------------
Component RightToken/annis/: 0 annnotations
Stats: nodes=84, avg_fan_out=0.71, max_fan_out=1, fan_out_99%=1, inv_fan_out_99%=8, max_depth=1
Implementation: DiskPathV1_D15
Status: "fully loaded"
------------
Component PartOf/annis/: 0 annnotations
Stats: nodes=115, avg_fan_out=0.99, max_fan_out=1, fan_out_99%=1, inv_fan_out_99%=26, max_depth=4
Implementation: DiskPathV1_D15
Status: "fully loaded"
------------

graphANNIS says good-bye!

----- stderr -----
